src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataBuilder.Heaps.cs

```diff
@@ -109,6 +127,10 @@ public MetadataBuilder(
                 throw new ArgumentException(SR.Format(SR.ValueMustBeMultiple, BlobUtilities.SizeOfGuid), nameof(guidHeapStartOffset));
             }
 
+            _createBlobBuilderFunc = createBlobBuilderFunc ?? (capacity => new BlobBuilder(capacity));
+            _userStringBuilder = _createBlobBuilderFunc(4 * 1024);
```

Comment:
Is this 4k arbitrary or based on prior art? I'm fine with it, just curious why it was choosen.